### PR TITLE
Supportpagination

### PR DESCRIPTION
**Pull request by shirren**

### CODE REVIEW

File: src/Network/JSONApi/Pagination.hs [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+} deriving Show
+
+{- |
+Pagination strategies are commonly implemented by the server of which Page and Offset
+are commonly used.
+-}
+data Strategy = PageStrategy | OffsetStrategy
+
+{- |
+Helper function to beuild relative links for a collection of resources of type ResourceEntity.
```
Comment (ben-barnes) [inline src/Network/JSONApi/Pagination.hs]: Typo (s/beuild/build)
```suggestion
Helper function to build relative links for a collection of resources of type ResourceEntity.
```

File: src/Network/JSONApi/Pagination.hs [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+lastPageIndex :: Strategy -> Pagination -> Word
+lastPageIndex PageStrategy page = numberOfPagesInPageList page
+lastPageIndex OffsetStrategy page = numberOfPagesInPageList page - 1
+
+{- |
+Simple pattern matcher than translates a Strategy to a query string element name.
+-}
+strategyToQueryStringNumberKey :: Strategy -> String
+strategyToQueryStringNumberKey PageStrategy = "page[number]"
```
Comment (ben-barnes) [inline src/Network/JSONApi/Pagination.hs]: I just wanted to double-check that these get URL-encoded somewhere?

Comment (ben-barnes) [reply src/Network/JSONApi/Pagination.hs]: Never mind - saw the URL encoding in your tests.

**Comment by shirren** [comment]
> Looking good! trophy

Thanks @toddmohney . PR is now ready for review, I wanted some feedback from a colleague of mine (@ben-barnes ). By the way, do you know why CI is stuck?

**Comment by shirren** [comment]
@toddmohney I am sure you are quite busy. But I was wondering if you had time to merge and publish the new version of the package by next week? Be great if we could start using the paging next week. Thanks in advance.

**Comment by toddmohney** [comment]
@shirren I plan on getting this merged and released this weekend. Thanks for your contribution! 

If you want to put this to work before it gets released, you can configure the `extra-deps` section of your `stack.yaml` file to reference a Git commit as described [here](https://docs.haskellstack.org/en/stable/yaml_configuration/#extra-deps)

**Comment by shirren** [comment]
> 
> 
> @shirren I plan on getting this merged and released this weekend. Thanks for your contribution!
> 
> If you want to put this to work before it gets released, you can configure the `extra-deps` section of your `stack.yaml` file to reference a Git commit as described [here](https://docs.haskellstack.org/en/stable/yaml_configuration/#extra-deps)

Thanks for the tip @toddmohney. Also thanks for taking the time out to merge on your weekend. We've also raised another issue. Keen to get your thoughts on it. Issue #34.

**Comment by shirren** [comment]
@toddmohney noticed that the PR has not been merged yet? Can I help do this?

**Comment by toddmohney** [comment]
@shirren thanks for you contribution and your patience. I've merged the PR and push the latest changes to Hackage https://hackage.haskell.org/package/json-api. The documentation is building now. I'll keep an eye on it an make sure it succeeds.

Cheers!

**Pull request merged**